StorageManagementSystem/Client/StorageManagerClient.py

```diff
@@ -61,31 +56,40 @@ def getFilesToStage( lfnList, jobState = None ):
     else:
       userName = None
       userGroup = None
-    result = _checkFilesToStage( seToLFNs, onlineLFNs,  # pylint: disable=unexpected-keyword-arg
+    # Check whether files are Online or Offline, or missing at SE
+    result = _checkFilesToStage( seToLFNs, onlineLFNs, offlineLFNs, absentLFNs,  # pylint: disable=unexpected-keyword-arg
                                  proxyUserName = userName,
                                  proxyUserGroup = userGroup,
                                  executionLock = True )
+
     if not result['OK']:
       return result
-    offlineLFNs = set( lfnList ) - onlineLFNs
+    failedLFNs = set( lfnList ) - onlineLFNs - set( offlineLFNs ) - set( absentLFNs )
 
-    for offlineLFN in offlineLFNs:
-      ses = lfnListReplicas[offlineLFN].keys()
+    for lfn in offlineLFNs:
+      ses = offlineLFNs[lfn]
       if ses:
-        offlineLFNsDict.setdefault( random.choice( ses ), list() ).append( offlineLFN )
+        offlineLFNsDict.setdefault( random.choice( ses ), list() ).append( lfn )
 
-  return S_OK( {'onlineLFNs':list( onlineLFNs ), 'offlineLFNs': offlineLFNsDict} )
+  return S_OK( {'onlineLFNs':list( onlineLFNs ), 'offlineLFNs': offlineLFNsDict, 'failedLFNs':list( failedLFNs ), 'absentLFNs':absentLFNs} )
 
 @executeWithUserProxy
-def _checkFilesToStage( seToLFNs, onlineLFNs ):
+def _checkFilesToStage( seToLFNs, onlineLFNs, offlineLFNs, absentLFNs ):
   """
   Checks on SEs whether the file is NEARLINE or ONLINE
   onlineLFNs is modified to contain the files found online
   """
   # Only check on storage if it is a tape SE
   failed = {}
   for se, lfnsInSEList in seToLFNs.iteritems():
-    fileMetadata = StorageElement( se ).getFileMetadata( lfnsInSEList )
+    seObj = StorageElement( se )
+    status = seObj.getStatus()
+    if not status['OK']:
+      gLogger.error( "Could not get SE status", "%s - %s" % ( se, status['Message'] ) )
+      return status
```

Comment:
Well, what can we do if we don't know the status of that SE? In practice this should never happen, right, unless the service is down...